apps/aecore/src/aec_headers.erl

```diff
@@ -272,10 +275,37 @@ validate_pow({#header{nonce        = Nonce,
             {error, incorrect_pow}
     end.
 
--spec validate_time({header(), aec_governance:protocols()}) ->
-                           ok | {error, block_from_the_future}.
-validate_time({#header{time = Time}, _}) ->
-    MaxAcceptedTime = aeu_time:now_in_msecs() + ?ACCEPTED_FUTURE_KEY_BLOCK_TIME_SHIFT,
+-spec validate_median_time({header(), aec_governance:protocols()}) ->
+        ok | {error, block_from_the_past}.
+validate_median_time({Header = #header{time = Time}, _}) ->
+    case aec_chain_state:median_timestamp(Header) of
+        {ok, MTime} when Time > MTime -> ok;
+        {ok, _MTime}                  -> {error, block_from_the_past};
+        error                         -> ok %% We can't know yet - checked later
+    end.
+
+-spec validate_micro_block_cycle_time({header(), aec_governance:protocols()}) ->
+        ok | {error, bad_micro_block_interval}.
+validate_micro_block_cycle_time({#header{time = Time, prev_hash = PrevHash}, _}) ->
+    case aec_chain:get_header(PrevHash) of
+        {ok, PrevHeader = #header{ time = PrevTime }} ->
+            MinAccepted =
+                case aec_headers:type(PrevHeader) of
+                    micro -> PrevTime + aec_governance:micro_block_cycle();
```

Comment:
I see lack of +1 here may be surprising [when reading the doc](https://github.com/aeternity/protocol/pull/167#discussion_r210232363).

Reply:
Though it looks to me this shall be more precisely specified in doc - rather than changing code.

Reply:
Yes, it is a matter of taste if we want to accept X ms difference or if we want to require X+1 ms...